### DISCUSSION TITLE
Celery memory leak with Airflow healthcheck - different health check?

### DISCUSSION THREAD

**Discussion by mtraynham** [celery/celery#7327] [Q&A] [answered]
Hello, 

I've observed a Celery memory leak when running Airflow 2.2.4/Celery 5.2.3  in a Docker container.  I've tracked it down to the health check that Airflow recommends for testing if the worker is still alive.  It uses the `celery inspect ping` command roughly every 10 seconds.  Over an hour of the Airflow app being idle, it increases the memory footprint of the service about 45 MB.

https://github.com/apache/airflow/blob/main/docs/apache-airflow/start/docker-compose.yaml#L132-L134

```
celery --app airflow.executors.celery_executor.app inspect ping -d "celery@$${HOSTNAME}"
```

I had opened https://github.com/apache/airflow/issues/21026 initially, which was closed as there were some potential suggestions of upgrading Celery from 5.2.2 to 5.2.3 would correct some memory leaks, and had referenced some commentary in https://github.com/celery/celery/issues/4843.  That however did not seem to be the case and I'm still observing the issue. 

I believe this particular variant of health check is referenced on [StackOverflow](https://stackoverflow.com/a/41377112), but I don't see it in the [Celery monitoring docs](https://docs.celeryproject.org/en/stable/userguide/monitoring.html).

Is there potentially a different Celery health check method or suggested solution for monitoring if the service is up?

I realize this is somewhat Airflow related, but at best, all they could do is change their documentation.  For more info on recreating the issue, you can see the Airflow bug about setup and recreation.

**Comment by pawl** [comment, marked as answer]
This might be related to?: https://github.com/celery/celery/issues/6009

**Reply by mtraynham** [reply]
That does seem surprisingly related given the command structure, although I'm completely ignorant of how celery or kombu works 😅 .  I'll forgo adding a Celery bug then and just link the Airflow issue to it.

Per chance, do you have any suggested work arounds I could maybe recommend back to Airflow for documentation purposes?  At the moment, I've completely removed our health check to avoid the issue, but I'm ok with that.
